src/shop.ts

```diff
@@ -58,6 +62,39 @@ export class Shop {
       timestamp: new Date().getTime(),
     }
 
-    return await this.storage.store(order)
+    return await this.storage.store(order, this.configuration.publicKey)
+  }
+
+  // back-end
+
+  async list(privateKey: JsonWebKey): Promise<Order[]> {
+    return this.storage.list(privateKey)
+  }
+
+  // setup
+
+  static async generateCrypto(): Promise<ShopCrypto> {
+    const crypto = new WebCrypto()
+    const pair = await crypto.generateKeyPair()
+
+    const privateKey = await crypto.exportKey(pair.privateKey)
+    const publicKey = await crypto.exportKey(pair.publicKey)
+    const id = await crypto.hash(publicKey.n)
+
+    return { privateKey, publicKey, id }
+  }
+
+  static generateConfiguration(publicKey: JsonWebKey, id: string): string {
```

Comment:
Same here (no need to have a dependency to `Shop` class by using a `static` function)
```
export generateShopConfiguration = async (): string => {...}
```

Reply:
See suggestion above. :)